From f6f03510d6c19e6009c903fed00294813126db3b Mon Sep 17 00:00:00 2001
Message-Id: <f6f03510d6c19e6009c903fed00294813126db3b.1664460024.git.github@sicherha.de>
In-Reply-To: <db11436c0ccbfee0f617ae7498c25fb2f4e76626.1664460024.git.github@sicherha.de>
References: <db11436c0ccbfee0f617ae7498c25fb2f4e76626.1664460024.git.github@sicherha.de>
From: Rui Ueyama <ruiu@bluewhale.systems>
Date: Thu, 29 Sep 2022 15:41:28 +0800
Subject: [PATCH 2/2] [ELF] Fix test for PPC64LE

`readelf --symbols` print out auxiliary information after a symbol name
for PPC64, so something like `foo$` didn't match.

     7: 0000000000000000    80 FUNC    GLOBAL DEFAULT    3 main [<localentry>: 8]
     8: 0000000000000000     0 NOTYPE  GLOBAL DEFAULT  UND .TOC.
     9: 0000000000000000    72 FUNC    GLOBAL DEFAULT   11 foo  [<localentry>: 8]
    10: 0000000000000000    40 FUNC    GLOBAL DEFAULT   19 bar

Fixes https://github.com/rui314/mold/issues/739
---
 test/elf/relocatable-archive.sh | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/test/elf/relocatable-archive.sh b/test/elf/relocatable-archive.sh
index e5544bb8..759cd871 100755
--- a/test/elf/relocatable-archive.sh
+++ b/test/elf/relocatable-archive.sh
@@ -38,8 +38,8 @@ ar crs $t/e.a $t/a.o $t/b.o $t/c.o
 ./mold -r -o $t/f.o $t/d.o $t/e.a
 
 readelf --symbols $t/f.o > $t/log
-grep -q 'foo$' $t/log
-grep -q 'bar$' $t/log
-! grep -q 'baz$' $t/log || false
+grep -q 'foo\b' $t/log
+grep -q 'bar\b' $t/log
+! grep -q 'baz\b' $t/log || false
 
 echo OK
-- 
2.37.3

